tests/native_api_tests/test_api_model.cpp

```diff
@@ -513,4 +538,133 @@ TEST_CASE("API Model") {
     }
 }
 
+/*
+
+source_1 -- node_0 --load_2
+
+source and node inputs are row based.
+load input is either row based or columnar.
+load update is row based / columnar and densse / sparse.
+
+invalid_id_t tests are for testing the error handling of the model when the id is not found in the update dataset.
+optional_id_t tests are for testing the model when the id is not added to the update dataset.
+
+*/
+TEST_CASE_TEMPLATE(
```

Comment:
Just commenting on the first line so it doesn't get too large...

This test is genius, thanks for such a big upgrade here, great effort. 

Personal request: please share in the tech review, I think it is worth it, we can all learn good stuff from here.